### PR TITLE
ROX-18162: Delete Network Graph 1.0 reducers and sagas

### PR DESCRIPTION
**Pull request by pedrottimark**
### Analysis

Search for action or selector calls outside of sagas.

#### clusters reducer

1. networkSagas has only `fetchClusters` action creator function call.

#### clusters sagas

2. clusterSagas has only `fetchClusters` service function call.

#### clusters service

3. normalize: see **Residue**

#### network reducers

1. OrchestratorComponentsToggle calls `updateNetworkNodes` action creator function.
2. searchAutocompleteSagas calls `getSelectedNetworkClusterId` selector function.

#### network sagas

1. networkPath
2. fetchClusters

#### pageSearch reducer

1. network search reducer has only import.

#### networkPath

1. integrationSagas imports networkPath

#### searchAutocompleteSagas

1. Only `getLocation` selector function call.

### Changes

Possible pattern for future deletion efforts:

1. Edit src/Containers/MainPage/Header/MastheadToolbar.tsx
    * Delete `useCase === useCases.NETWORK` from conditional rendering.
    * Delete `useCase` prop (see next item).

2. Edit OrchestratorComponentsToggle
    * Delete `useCase` prop and special case for Network Graph 1.0 container.
    * Delete `updateNetworkNodes` action creator function.

3. Delete obsolete files for networkSagas data model and cytoscape rendering:
    * src/utils/networkLink.utils.js
    * src/utils/networkNode.utils.js

4. Edit src/sagas/searchAutocompleteSagas.js
    * Delete special case for /main/network that had `getSelectedNetworkClusterId` selector function call and which did not use `networkPath` by the way!
    * Delete `getLocation` selector function call.

5. Delete network sagas and reducers:
    * `git rm ui/apps/platform/src/sagas/networkSagas.js`
    * `git rm -r ui/apps/platform/src/reducers/network`

6. Delete cluster sagas and reducers:
    * `git rm ui/apps/platform/src/sagas/clusterSagas.js`
    * `git rm ui/apps/platform/src/reducers/clusters.js`

7. Edit src/services/ClustersService.ts
    * Delete `fetchClusters` and `fetchCluster` service functions.

8. Delete pageSearch helper file
    * `git rm ui/apps/platform/src/reducers/pageSearch.ts`

9. src/reducers/index.js

10. Edit src/sagas/index.js

11. Edit cypress/integration/networkGraphPF/networkGraph.helpers.js
    * Delete epoch request because was immediate with network sagas but now with polling interval 30 seconds **after** namespaces have been selected.

12. Edit src/sagas/integrationSagas.js
    * Delete special case for notifiers because of Network Graph.

13. Edit src/routePaths.js
    * Delete paths and title.

14. Bonus delete NETWORK from use cases:
    * src/constants/useCaseTypes.js
    * src/messages/useCase.js

### Residue

1. Follow up in ClustersService:
    * Rename `fetchClustersAsArray` function?
    * Remove `normalize` from `saveCluster` function? Then delete schemas.js and normalize package dependency!
2. Investigate routes reducer:
    * Is it obsolete because `getLocation` selector function has no more calls.
    * Are exports also obsolete because authSagas can just as well import directly from connected-react-router package?
3. Investigate possibly obsolete network files:
    * src/constants/networkFilterModes.js
    * src/constants/networkGraph.js
    * src/utils/networkGraphUtils.js

## Checklist
- [x] Investigated and inspected CI test results
- [x] Edited integration tests

## Testing Performed

TL;DR Apparently webpack is able to move some code out of main bundle after we delete global references in reducers and sagas.

1. `yarn lint` in ui
2. `yarn build` in ui
    * `wc build/static/js/*.js`
        branch - master total: -404 = 11097466 - 11097870
        branch - master main.js: -151947 = 4697502 - 4849449
    * `ls -al build/static/js/*.js | wc`
        branch - master: 4 = 78 - 74 files
3. `yarn start` in ui

### Manual testing

1. Visit /main/dashboard

2. Click **Network Graph** in left navigation to visit /main/network-graph

3. Continue to visit every link in left navigation

### Integration testing

1. `yarn cypress-open` in ui/apps/platform
    * networkGraphPF/networkDeploymentSidebar.test.js
    * networkGraphPF/networkGraphPF.test.js

### CODE REVIEW

File: ui/apps/platform/src/reducers/pageSearch.ts [hunk 1]
```diff
@@ -1,150 +0,0 @@
...
-export function getSelectors<EntityPrefix extends string>(
-    prefix: EntityPrefix
-): SearchSelectors<EntityPrefix> {
-    return {
-        [`get${capitalize(prefix)}SearchOptions`]: (state: SearchState) => state.searchOptions,
-        [`get${capitalize(prefix)}SearchModifiers`]: (state: SearchState) => state.searchModifiers,
-        [`get${capitalize(prefix)}SearchSuggestions`]: (state: SearchState) =>
-            state.searchSuggestions,
-    } as SearchSelectors<EntityPrefix>;
-}
```
Comment (alwayshooin) [inline ui/apps/platform/src/reducers/pageSearch.ts:150]: oh was this the last instance of using the page search reducers?

File: ui/apps/platform/src/routePaths.js [hunk 2]
```diff
@@ -11,8 +11,6 @@ export const authResponsePrefix = '/auth/response/';
 export const authorizeRoxctlPath = '/authorize-roxctl';
 
 export const dashboardPath = `${mainPath}/dashboard`;
-export const networkBasePath = `${mainPath}/network`;
-export const networkPath = `${networkBasePath}/:deploymentId?/:externalType?`;
 export const networkBasePathPF = `${mainPath}/network-graph`;
```
Comment (alwayshooin) [inline ui/apps/platform/src/routePaths.js:14]: should we just rename this to `networkBasePath` since there is only one version of the network graph now?

File: ui/apps/platform/src/sagas/clusterSagas.js [hunk 3]
```diff
@@ -1,21 +0,0 @@
...
-    }
-}
-
-function* watchFetchRequest() {
-    yield takeLatest(types.FETCH_CLUSTERS.REQUEST, getClusters);
-}
-
-export default function* clusters() {
-    yield all([fork(watchFetchRequest)]);
-}
```
Comment (alwayshooin) [inline ui/apps/platform/src/sagas/clusterSagas.js:21]: do we not use this when we're on the `clusters` page?

File: ui/apps/platform/src/services/ClustersService.ts [hunk 4]
```diff
@@ -147,17 +133,6 @@ export function upgradeClusters(ids = []): Promise<Empty[]> {
...
- * Fetches cluster by its ID.
- */
-// TODO specify return type after we rewrite without normalize
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export function fetchCluster(id: string) {
-    return axios.get(`${clustersUrl}/${id}`).then((response) => ({
-        response: normalize(response.data, { cluster: clusterSchema }),
-    }));
-}
-
```
Comment (alwayshooin) [inline ui/apps/platform/src/services/ClustersService.ts:160]: do we just not use this and instead use `fetchClusterWithRetentionInformation` now?

File: ui/apps/platform/src/sagas/clusterSagas.js (hunk 3 above)
Comment (pedrottimark) [reply ui/apps/platform/src/sagas/clusterSagas.js:21]: Good question: It calls `fetchClustersWithRetentionInfo` service function.
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Clusters/ClustersTablePanel.js#L123-L134

File: ui/apps/platform/src/services/ClustersService.ts (hunk 4 above)
Comment (pedrottimark) [reply ui/apps/platform/src/services/ClustersService.ts:160]: Yes, indeed. The other service function has additional information.
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Clusters/ClustersSidePanel.tsx#L131

File: ui/apps/platform/src/reducers/pageSearch.ts (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/reducers/pageSearch.ts:150]: Yes, no more page search reducers.

On the other side of the coin, `URLSearchInput` still uses searchAutocomplete reducer and sagas.

File: ui/apps/platform/cypress/integration/networkGraphPF/networkGraph.helpers.js [hunk 5]
```diff
@@ -63,18 +63,15 @@ export function selectDeployment(deployment) {
 
 export const notifiersAlias = 'notifiers';
 export const clustersAlias = 'clusters';
-export const networkPoliciesGraphEpochAlias = 'networkpolicies/graph/epoch';
+// Removed the following because request has 30 second delay from polling interval:
```
Comment (dvail) [inline ui/apps/platform/cypress/integration/networkGraphPF/networkGraph.helpers.js:66]: 👍 

FWIW we also receive the `epoch` value from the main request that retrieves the network graph info once a cluster is selected. This value is currently unimplemented (always returns `0`), but if/when it is implemented we might be able to use it as a reliable way to tell that the graph has "settled".

**Review by dvail — APPROVED** [review]
